src/index.js

```diff
@@ -0,0 +1,261 @@
+
+const getOtherContents = ({ installation, usage, tests }) => {
+  const contents = [];
+  if (installation) contents.push("- [Installation](#installation)");
+  if (usage) contents.push("- [Usage](#usage)");
+  if (tests) contents.push("- [Tests](#tests)");
+  return contents;
+};
+
+// constructing the dynamic README.md
+// project title
+const constructTitle = (projectTitle, license) => {
+  return `# ${projectTitle} ![${license}](https://img.shields.io/static/v1?label=${license}&message=Licence&color=<COLOR>)`;
+};
+
+// table of contents
+const generateTableOfContents = (installation, usage, tests) => {
+  const contents = [
+    "- [Description](#description)",
+    ...getOtherContents(installation, usage, tests),
+    "- [Contributing](#contributing)",
+    "- [License](#license)",
+    "- [Question](#question)",
+  ];
+  return `## Table of Contents\n
+${contents.join("\n")}
+`;
+};
+
+// project description
+const constructDesc = (projectDesc) => {
+  return `
+## Description
+
+${projectDesc}`;
+};
+
+// project installation
+const constructInstall = (installation) => {
+  if (installation) {
+    return `
+## Installation\n
+Follow these steps for installation:
+
+${installation
+  .map(function (installSteps) {
+    return `
+\`\`\`
+${installSteps.hasInstallScript}
+\`\`\``;
+  })
+  .join("\n")}`;
+  } else {
+    return "";
+  }
+};
+
+// project directions
+const constructDirections = (usage) => {
+  if (usage) {
+    return `
+## Usage
+
+${usage
+  .map(function (usageSteps) {
+    return `
+\`\`\`
+${usageSteps.hasUsage}
+\`\`\``;
+  })
+  .join("\n")}`;
+  } else {
+    return "";
+  }
+};
+
+// test table
+const constructTestingTable = (projectAnswer) => {
+  return `## Testing\n
+  <table>
+<tbody><tr>
+<th>Browser</th>
+<th>Working Links</th>
+<th>Formatting Correctly</th>
+<th>Notes</th> 
+</tr>
+<tr>
+<th>Chrome</th>
+<td>${projectAnswer.chromeLinks}</td>
+<td>${projectAnswer.chrome}</td>
+<td>${projectAnswer.chromeNotes}</td>
+</tr>
+<tr>
+<th>FireFox</th>
+<td>${projectAnswer.firefoxLinks}</td>
+<td>${projectAnswer.firefox}</td>
+<td>${projectAnswer.firefoxNotes}</td>
+</tr>
+<tr>
+<th>Internet Explorer</th>
+<td>${projectAnswer.explorerLinks}</td>
+<td>${projectAnswer.explorer}</td>
+<td>${projectAnswer.explorerNotes}</td>
+</tr>
+</tbody></table>`;
+};
+
+// further tests
+const constructTest = (tests) => {
+  if (tests) {
+    return `
+## Further Testing
+
+${tests
+  .map(function (testSteps) {
+    return `
+\`\`\`
+${testSteps.hasTests}
+\`\`\``;
+  })
+  .join("\n")}`;
+  } else {
+    return "";
+  }
+};
+
+// project license
+const constructLicense = (license) => {
+  return `## License\n
+  
+  This project is licensed under the terms of the ${license} license.`;
+};
+
+// project questions
+const constructQuestion = (gitHubUserName, email) => {
+  return `## Questions\n
+  If you have any questions about this project, or would like further information please contact me via:\n
+  GitHub: ${gitHubUserName}\n
```

Comment:
The GitHub username should be a link. This can be done by using square brackets and then the link in brackets immediately after `[visible text here](link)`